common/src/main/java/com/wynntils/core/managers/Manager.java

```diff
@@ -4,15 +4,20 @@
  */
 package com.wynntils.core.managers;
 
+import com.wynntils.core.WynntilsMod;
+import java.util.List;
+
 /**
- * Managers can have two static methods:
- * <p>
- * init: Called when manager is enabled
- * <p>
- * disable: Called when manager is disabled
- * <p>
- * Managers are automatically registered to event bus, use static event methods.
- * <p>
- * See {@link Model} and {@link CoreManager} for the different manager types.
- * */
-public abstract class Manager {}
+ * Managers constitute the core functionality of Wynntils. They are created
+ * as singletons in the {@link Managers} holding class.
+ */
+public abstract class Manager {
+    protected Manager(List<Manager> dependencies) {
+        // dependencies are technically not used, but only required
```

Comment:
Oh. I see. Well, this is kinda weird isn't it?

Reply:
I am using Java:s built-in syntax check to make sure we get bootstrapping in the correct order.

If we really wanted, we could of course set up a rule that no Manager might read Managers.<mgr>, and only access the managers they need from the one they get through the constructor. However, I think it just adds boilerplate, and make accesses to other Managers from within a Manager looking quite different from the rest of the code.